src/esw/brushless.py

```diff
@@ -212,13 +216,100 @@ async def update(self) -> None:
+                state=[self._motor_bridge_by_name[name].moteus_state.state for name in self._motor_names],
+                error=[self._motor_bridge_by_name[name].moteus_state.error_name for name in self._motor_names],
+            ),
+        )
+
+    def update_time(self) -> None:
+        """
+        Updates the last updated time
+        """
+        self._last_updated_time = t.time()
+
+    async def run(self) -> None:
+        """
+        Constantly updates the bridges.
+        """
+        previously_lost_communication = True
+        while not rospy.is_shutdown():
+            for name, bridge in self._motor_bridge_by_name.items():
+                time_diff_since_updated = t.time() - self._last_updated_time
+                lost_communication = (
+                    time_diff_since_updated > MotorsManager.BASESTATION_TO_ROVER_NODE_WATCHDOG_TIMEOUT_S
+                )
+                if lost_communication:
+                    if not previously_lost_communication:
+                        rospy.loginfo("Lost communication")
+                        previously_lost_communication = True
+                    bridge.set_command(
+                        CommandData(
+                            position=CommandData.POSITION_FOR_VELOCITY_CONTROL,
+                            velocity=CommandData.ZERO_VELOCITY,
+                            torque=CommandData.DEFAULT_TORQUE,
+                        )
+                    )
+                elif previously_lost_communication:
+                    previously_lost_communication = False
+                    rospy.loginfo("Regained communication")
+
+                await bridge.update()
+
+                index = self._motor_names.index(name)
+
+                self._motors_status.joint_states.position[index] = bridge.moteus_data.position
+                self._motors_status.joint_states.velocity[index] = bridge.moteus_data.velocity
+                self._motors_status.joint_states.effort[index] = bridge.moteus_data.torque
+                self._motors_status.moteus_states.state[index] = bridge.moteus_state.state
+                self._motors_status.moteus_states.error[index] = bridge.moteus_state.error_name
+
+            self._motors_status_publisher.publish(self._motors_status)
+
+    def update_command_data(self, command_data_list: List[CommandData]) -> None:
+        """
+        Updates the commands
+        :param command_data_list: Has an array of command data that represents the commands for each motor.
+        """
+        assert len(command_data_list) == len(self._motor_names)
+        for i, command_data in enumerate(command_data_list):
+            motor_name = self._motor_names[i]
+            if self._motor_bridge_by_name[motor_name].moteus_state.state == MoteusState.ARMED_STATE:
+                self._motor_bridge_by_name[motor_name].set_command(command_data)
+
+
+class ArmManager:
+    ARM_NAMES = ["joint_a", "joint_c", "joint_d", "joint_e", "joint_f", "gripper"]
```

Comment:
Can't we get all these from the config file?

Reply:
Yup, I made changes